### PR TITLE
Update batch calculations

### PR DESCRIPTION
**Pull request by manchen07**
I added aggregate and weighting arguments and replaced `do()` with `summarise()`. Aggregating is done for long format and then converted to wide format if required. Need to discuss this more. Also need to discuss how to aggregate confidence intervals.

### CODE REVIEW

File: R/calculate-effect-sizes.R [hunk 1]
```diff
@@ -341,10 +355,11 @@ batch_calc_ES <- function(dat,
   
   if (!is.null(session_number)) dat <- dplyr::arrange(dat, !!rlang::sym(session_number))
   
-  dat %>%  
+  ES_ests_long <- 
+    dat %>% 
     dplyr::group_by(!!!rlang::syms(grouping)) %>%
-    dplyr::do(calc_ES(condition = .data[[condition]], 
-                      outcome = .data[[outcome]], 
+    dplyr::summarise(calc_ES(condition = .data[[condition]],
```
Comment (jepusto) [inline R/calculate-effect-sizes.R:361]: Well that simplifies things!

File: R/calculate-effect-sizes.R [hunk 2]
```diff
@@ -353,9 +368,91 @@ batch_calc_ES <- function(dat,
...
+        ) 
+    }
+    
+    result <- ES_ests
+    
+  } else {
+    
+    Est <- SE <- n <- CI_lower <- CI_upper <- NULL
+    
+    if (weighting == "1/V") {
```
Comment (jepusto) [inline R/calculate-effect-sizes.R:396]: Handle weights by first computing the weighting variable (depending on the `weighting` argument), then summarizing using weighted averages.

File: R/calculate-effect-sizes.R [hunk 3]
```diff
@@ -353,9 +368,91 @@ batch_calc_ES <- function(dat,
...
+        dplyr::summarise(
+          Est = mean(Est),
+          n = n(),
+          SE = sqrt(sum(SE^2)) / n) %>% 
+        dplyr::select(-n)
+    }
+    
+    if (!is.null(confidence)) {
+      
+      res <-
```
Comment (jepusto) [inline R/calculate-effect-sizes.R:416]: This is fine for handling CI calculations with aggregated effect sizes.

File: R/calculate-effect-sizes.R [hunk 4]
```diff
@@ -353,9 +368,91 @@ batch_calc_ES <- function(dat,
...
+          names_glue = "{ES}_{.value}",
+          values_from = - c(!!!grouping, ES)
+        ) 
+    }
+    
+    result <- ES_ests
+    
+  } else {
+    
+    Est <- SE <- n <- CI_lower <- CI_upper <- NULL
```
[the remark this answers is not in the document]
Comment (manchen07) [reply R/calculate-effect-sizes.R:394]: This is used for handling the "no visible binding for global variable" in R CMD check (https://www.r-bloggers.com/2019/08/no-visible-binding-for-global-variable/).

File: R/calculate-effect-sizes.R [hunk 5]
```diff
@@ -1,3 +1,55 @@
+if(getRversion() >= "2.15.1")  utils::globalVariables(c("ES", "Est", "SE", "CI_upper", "CI_lower"))
```
Comment (manchen07) [inline R/calculate-effect-sizes.R:1]: This is used to handle the warning message "no visible binding for global variable" in R CMD check.

**Review by jepusto — CHANGES_REQUESTED** [review]
I made a few small tweaks on this branch. Three issues for your consideration:

### Syntax for `aggregate` 

Right now `aggregate`  will have to be a subset of the variables in `grouping` in order to work. The syntax is also the opposite of what I had in mind. I think it will be more intuitive if we 

1. Calculate ES estimates grouping by _both_ `grouping` and `aggregate`
2. Summarize the ES estimates and SEs _over_ the levels of `aggregate`, while grouping by `grouping`  (and `ES`).

So for example, in the `Schmidt2007` data, if we want one ES for each `Behavior_type` and `Case_pseudonym`, with effects calculated for each `Phase_num` and the averaged together, the syntax would be
```
  ES_agg_CI <- 
    batch_calc_ES(dat = Schmidt2007,
                  grouping = c(Behavior_type, Case_pseudonym),
                  aggregate = Phase_num,
                  weighting = "1/V",
                  condition = Condition,
                  outcome = Outcome,
                  ES = c("LRRi","LRRd", "Tau"),
                  improvement = direction,
                  scale = "count",
                  bias_correct = TRUE,
                  confidence = .95,
                  format = "long")
```

I've modified the unit tests to use the desired syntax. Please update the code to pass the tests.

### `Weighting` specification

Right now, the error handling for the `weighting` argument says that it "must be a variable name or a string specifying '1/V' or 'equal'." But I think it should have to be one of the allowed strings. Right now, if you specify a variable name I think it will default to equal weighting, which doesn't seem right. I've added a unit test to this effect.

### Example with `aggregate`

Please add an example to the documentation demonstrating use of the `aggregate` argument.

**Review by jepusto — APPROVED** [review]

**Pull request merged**